check_whitespace.c

```diff
@@ -67,7 +67,13 @@ int is_clean(char* str) {
   // greater than the second.
   result = strcmp(str, cleaned);
 
-  return result == 0;
+  if(strcmp("",cleaned)){
```

Comment:
You totally had the right idea here, but got the test wrong. You needed to avoid `free`ing if `cleaned` _is_ the empty string. `strcmp` returns 0 if the two strings are the same, and `if` sees zero as `false`.

Negating the test fixes things and you get no `valgrind` warnings:

```suggestion
  if(!strcmp("",cleaned)){
```